SourceCode/AgOpenGPS.Core/Streamers/Field/BoundaryStreamer.cs

```diff
@@ -0,0 +1,105 @@
+﻿using AgOpenGPS.Core.Interfaces;
+using AgOpenGPS.Core.Models;
+using System;
+using System.IO;
+
+namespace AgOpenGPS.Core.Streamers
+{
+    public class BoundaryStreamer : FieldAspectStreamer
+    {
+        public BoundaryStreamer(
+            ILogger logger
+        ) :
+            base(logger, "Boundary.txt")
+        {
+        }
+
+        public Boundary TryRead(string fieldPath)
```

Comment:
Usually, `Try...` methods have a slightly different signature:
```suggestion
        public bool TryRead(string fieldPath, out Boundary boundary)
```
That way you can check as a caller if the read was successful or not.

Reply:
I was not aware of the more common signature. Anyway the caller will ignore the return value. Besides that, some of the AspectsStreamers can also return null as a valid result and that does not work very well with an out parameter.

Reply:
Okay, different question: is it really necessary, that theses classes even provide the `Try...` methods? For me the `try`-`catch` and the error handling seems to belong to the caller of those methods.
This would have several advantages:
- The AspectStreamers get simpler
- They do not need to know about logging or the presenter

Reply:
It is not really necessary of course. A different approach would be possible, but I don't want to bother the caller with various error states, like missing or corrupt files. So I prefer to do the logging here. Also this code will not only be used during openening and closing a field, but also in during editing of the Flags or recording paths, generation of tram lines and also during field selection or copying of fields, and I don't want to bother all these callers either.